metadata-ingestion/src/datahub/ingestion/source/powerbi/rest_api_wrapper/data_resolver.py

```diff
@@ -0,0 +1,660 @@
+                name=raw_instance["name"],
+                displayName=raw_instance.get("displayName"),
+                order=raw_instance.get("order"),
+            )
+            for raw_instance in response_dict["value"]
+        ]
+
+    def get_users(self, workspace_id: str, entity: str, entity_id: str) -> List[User]:
+        return []  # User list is not available in regular access
+
+
+class AdminAPIResolver(DataResolverBase):
+
+    # Admin access endpoints
+    API_ENDPOINTS = {
+        Constant.DASHBOARD_LIST: "{POWERBI_ADMIN_BASE_URL}/groups/{WORKSPACE_ID}/dashboards",
+        Constant.TILE_LIST: "{POWERBI_ADMIN_BASE_URL}/dashboards/{DASHBOARD_ID}/tiles",
+        Constant.REPORT_LIST: "{POWERBI_ADMIN_BASE_URL}/groups/{WORKSPACE_ID}/reports",
+        Constant.SCAN_GET: "{POWERBI_ADMIN_BASE_URL}/workspaces/scanStatus/{SCAN_ID}",
+        Constant.SCAN_RESULT_GET: "{POWERBI_ADMIN_BASE_URL}/workspaces/scanResult/{SCAN_ID}",
+        Constant.SCAN_CREATE: "{POWERBI_ADMIN_BASE_URL}/workspaces/getInfo",
+        Constant.ENTITY_USER_LIST: "{POWERBI_ADMIN_BASE_URL}/{ENTITY}/{ENTITY_ID}/users",
+        Constant.DATASET_LIST: "{POWERBI_ADMIN_BASE_URL}/groups/{WORKSPACE_ID}/datasets",
+    }
+
+    def create_scan_job(self, workspace_id: str) -> str:
+        """
+        Create scan job on PowerBI for the workspace
+        """
+        request_body = {"workspaces": [workspace_id]}
+
+        scan_create_endpoint = AdminAPIResolver.API_ENDPOINTS[Constant.SCAN_CREATE]
+        scan_create_endpoint = scan_create_endpoint.format(
+            POWERBI_ADMIN_BASE_URL=DataResolverBase.ADMIN_BASE_URL
+        )
+
+        logger.debug(
+            f"Creating metadata scan job, request body {request_body}",
+        )
+
+        res = self._request_session.post(
+            scan_create_endpoint,
+            data=request_body,
+            params={
+                "datasetExpressions": True,
+                "datasetSchema": True,
+                "datasourceDetails": True,
+                "getArtifactUsers": True,
+                "lineage": True,
+            },
+            headers={Constant.Authorization: self.get_access_token()},
+        )
+
+        res.raise_for_status()
+        # Return scan_id of Scan created for the given workspace
+        scan_id = res.json()["id"]
+
+        logger.info(f"Scan id({scan_id})")
+
+        return scan_id
+
+    @staticmethod
+    def _calculate_max_retry(minimum_sleep: int, timeout: int) -> int:
+        if timeout < minimum_sleep:
+            logger.info(
+                f"Setting timeout to minimum_sleep time {minimum_sleep} seconds"
+            )
+            timeout = minimum_sleep
+
+        return timeout // minimum_sleep
+
+    def wait_for_scan_to_complete(self, scan_id: str, timeout: int) -> Any:
+        """
+        Poll the PowerBi service for workspace scan to complete
+        """
+        minimum_sleep = 3
```

Comment:
what is the time unit? I recommend putting it in the name:

minimum_sleep_ms or s or whatever

Reply:
done